File: studyOC/studyOC/App/rootViewController.m
//
//  rootViewController.m
//  studyOC
//
//  Created by Hahn on 2021/9/15.
//

#import "rootViewController.h"
#import "LinkViewController.h"
#import "LinkModel.h"

@interface rootViewController ()

@end

@implementation rootViewController

- (void)dealloc {
    [[NSNotificationCenter defaultCenter] removeObserver:self name:@"Scheme" object:nil];
    [[NSNotificationCenter defaultCenter] removeObserver:self name:@"Universal" object:nil];
}

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor blackColor];
    
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(Scheme:) name:@"Scheme" object:nil];
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(Universal:) name:@"Universal" object:nil];
}

- (void)Scheme:(NSNotification *)noti {
    LinkModel *model = [LinkModel new];
    model = noti.object;
    LinkViewController *vc = [LinkViewController new];
    vc.model = model;
    [self.navigationController pushViewController:vc animated:YES];
}

- (void)Universal:(NSNotification *)noti {
    LinkModel *model = [LinkModel new];
    model = noti.object;
    LinkViewController *vc = [LinkViewController new];
    vc.model = model;
    [self.navigationController pushViewController:vc animated:YES];
}



@end

File: studySwift/studySwift/App/rootViewController.swift
//
//  rootViewController.swift
//  studySwift
//
//  Created by Hahn on 2021/9/16.
//

import UIKit

class rootViewController: UIViewController {

    deinit {
        NotificationCenter.default.removeObserver(self, name: NSNotification.Name.init(rawValue: "Scheme"), object: nil)
        NotificationCenter.default.removeObserver(self, name: NSNotification.Name.init(rawValue: "Universal"), object: nil)
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        NotificationCenter.default.addObserver(self, selector: #selector(Scheme), name: NSNotification.Name.init(rawValue: "Scheme"), object: nil)
        NotificationCenter.default.addObserver(self, selector: #selector(Universal), name: NSNotification.Name.init(rawValue: "Universal"), object: nil)
        
    }
    
    @objc func Scheme(notification: Notification) {
        let model: LinkModel = notification.object as! LinkModel
        let vc = LinkViewController()
        vc.model = model
        self.navigationController?.pushViewController(vc, animated: true)
    }
    
    @objc func Universal(notification: Notification) {
        let model: LinkModel = notification.object as! LinkModel
        let vc = LinkViewController()
        vc.model = model
        self.navigationController?.pushViewController(vc, animated: true)
    }

}
